flow/cmd/handler.go

```diff
@@ -154,7 +168,65 @@ func (h *FlowRequestHandler) CreateCDCFlow(
 		TypedSearchAttributes: shared.NewSearchAttributes(cfg.FlowJobName),
 	}
 
-	if err := h.createCdcJobEntry(ctx, req, workflowID); err != nil {
+	if err := h.createCdcJobEntry(ctx, req, workflowID, false); err != nil {
+		slog.ErrorContext(ctx, "unable to create flow job entry", slog.Any("error", err))
+		return nil, exceptions.NewInternalApiError(fmt.Errorf("unable to create flow job entry: %w", err))
+	}
+
+	if _, err := h.temporalClient.ExecuteWorkflow(ctx, workflowOptions, peerflow.CDCFlowWorkflow, cfg, nil); err != nil {
+		slog.ErrorContext(ctx, "unable to start PeerFlow workflow", slog.Any("error", err))
+		return nil, exceptions.NewInternalApiError(fmt.Errorf("unable to start PeerFlow workflow: %w", err))
+	}
+
+	return &protos.CreateCDCFlowResponse{
+		WorkflowId: workflowID,
+	}, nil
+}
+
+func (h *FlowRequestHandler) CreateCDCFlowManaged(
+	ctx context.Context, req *protos.CreateCDCFlowRequest,
+) (*protos.CreateCDCFlowResponse, error) {
+	cfg := req.ConnectionConfigs
+	internalVersion, err := internal.PeerDBForceInternalVersion(ctx, req.ConnectionConfigs.Env)
+	if err != nil {
+		return nil, exceptions.NewInternalApiError(fmt.Errorf("failed to get internal version: %w", err))
+	}
+	cfg.Version = internalVersion
+
+	if cfg.Resync {
+		return nil, exceptions.NewInvalidArgumentApiError(errors.New("resync is not supported in the managed API"))
+	}
+
+	workflowID := cfg.FlowJobName + "-peerflow"
```

Comment:
are we making wf IDs deterministic without the uuid suffix?

Reply:
One issue with having a new API is that we need to wait for all users to be up to date before switching the API. Instead if we add a new parameter, it would allow us to switch out of the box

Reply:
Might rename the existing endpoint then so the managed caller doesn't have to be deployed after